torch/_dynamo/exc.py

```diff
@@ -41,6 +41,14 @@ class RestartAnalysis(TorchDynamoException):
     pass
 
 
+class SpeculationRestartAnalysis(RestartAnalysis):
+    pass
+
+
+class UnspecializeRestartAnalysis(RestartAnalysis):
```

Comment:
Is this handled anywhere specifically? Or are you just using the type for more contextual info?

Reply:
It is just to give more context info.  Also it allows someone in the future to understand the reason for a restart.

Reply:
Actually turns out I did need to use this to a fix a bug, so I guess good forsight on my part...